src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -100,18 +103,92 @@ private void OnDestroy()
+                ZipFile.CreateFromDirectory(gameplaySessionScreenshotDirectory, gameplaySessionScreenshotDirectory + ".zip");
+                RGDebug.LogInfo($"Finished zipping replay to file: {gameplaySessionScreenshotDirectory}.zip");
+                
+                // Finally, we also save a thumbnail, by choosing the middle file in the screenshots
+                var screenshotFiles = Directory.GetFiles(gameplaySessionScreenshotDirectory);
+                var middleFile = screenshotFiles[screenshotFiles.Length / 2]; // this gets floored automatically
+                File.Copy(middleFile, thumbnailPath);
+                Directory.Delete(gameplaySessionScreenshotDirectory, true);
+            });
+            CreateAndUploadGameplaySession(endTime);
+        }
+
+        private async void CreateAndUploadGameplaySession(DateTime endTime)
+        {
+            RGDebug.LogInfo($"Creating and uploading GameplaySession on the backend, from {_startTime} to {endTime} with {_tickNumber} ticks");
+            try
+            {
+                
+                // First, create the gameplay session
+                long gameplaySessionId = -1;
+                await _rgServiceManager.CreateGameplaySession(_startTime, endTime, _tickNumber,
+                    (response) =>
+                    {
+                        gameplaySessionId = response.id;
+                        RGDebug.LogInfo($"Created gameplay session with id: {response.id}");
+                    },
+                    () => { RGDebug.LogError($"Failed to create gameplay session"); });
+                
+                // If the gameplay session was not created, return
+                if (gameplaySessionId == -1)
                 {
-                    RGDebug.LogInfo($"Zipping recording replay to file: {_currentVideoDirectory}.zip");
-                    ZipFile.CreateFromDirectory(theVideoDirectory, theVideoDirectory + ".zip");
-                    Directory.Delete(theVideoDirectory, true);
-                    RGDebug.LogInfo($"Finished zipping replay to file: {_currentVideoDirectory}");
-                });
-                _isRecording = false;
+                    return;
+                }
+                
+                // Upload the gameplay session data
+                await _rgServiceManager.UploadGameplaySessionData(gameplaySessionId,
+                    _currentGameplaySessionDirectoryPrefix + "_data.zip",
+                    () => { RGDebug.LogInfo($"Uploaded gameplay session data"); },
+                    () => { RGDebug.LogError($"Failed to upload gameplay session data"); });
+                
+                // Next, upload the gameplay session screenshots
+                await _rgServiceManager.UploadGameplaySessionScreenshots(gameplaySessionId,
+                    _currentGameplaySessionDirectoryPrefix + "_screenshots.zip",
+                    () => { RGDebug.LogInfo($"Uploaded gameplay session screenshots"); },
```

Comment:
We should include more information if we are going to log this message, like the path